test/multipart_test.dart

```diff
@@ -0,0 +1,249 @@
+    ];
+    var request = new http.Request.multipart(dummyUrl, files: files);
+
+    expect(request, bodyMatches('''
+        --{{boundary}}
+        content-type: text/plain; charset=utf-8
+        content-disposition: form-data; name="file"; filename="foo%22bar"
+
+        contents
+        --{{boundary}}--
+        '''));
+  });
+
+  test('with a string file with a content-type but no charset', () {
+    var files = [
+      new http.MultipartFile.fromString('file', '{"hello": "world"}',
+          contentType: new MediaType('application', 'json'))
+    ];
+    var request = new http.Request.multipart(dummyUrl, files: files);
+
+    expect(request, bodyMatches('''
+        --{{boundary}}
+        content-type: application/json; charset=utf-8
+        content-disposition: form-data; name="file"
+
+        {"hello": "world"}
+        --{{boundary}}--
+        '''));
+  });
+
+  test('with a file with a iso-8859-1 body', () {
+    // "Ã¥" encoded as ISO-8859-1 and then read as UTF-8 results in "å".
+    var files = [
+      new http.MultipartFile.fromString('file', 'non-ascii: "Ã¥"',
+          contentType:
+              new MediaType('text', 'plain', {'charset': 'iso-8859-1'}))
+    ];
+    var request = new http.Request.multipart(dummyUrl, files: files);
+
+    expect(request, bodyMatches('''
+        --{{boundary}}
+        content-type: text/plain; charset=iso-8859-1
+        content-disposition: form-data; name="file"
+
+        non-ascii: "å"
+        --{{boundary}}--
+        '''));
+  });
+
+  test('with a stream file', () {
+    var controller = new StreamController(sync: true);
+    var files = [new http.MultipartFile('file', controller.stream, 5)];
+    var request = new http.Request.multipart(dummyUrl, files: files);
+
+    expect(request, bodyMatches('''
+        --{{boundary}}
+        content-type: application/octet-stream
+        content-disposition: form-data; name="file"
+
+        hello
+        --{{boundary}}--
+        '''));
+
+    controller.add([104, 101, 108, 108, 111]);
+    controller.close();
+  });
+
+  test('with an empty stream file', () {
+    var controller = new StreamController(sync: true);
+    var files = [new http.MultipartFile('file', controller.stream, 0)];
+    var request = new http.Request.multipart(dummyUrl, files: files);
+
+    expect(request, bodyMatches('''
+        --{{boundary}}
+        content-type: application/octet-stream
+        content-disposition: form-data; name="file"
+
+
+        --{{boundary}}--
+        '''));
+
+    controller.close();
+  });
+
+  test('with a byte file', () {
+    var files = [
+      new http.MultipartFile.fromBytes('file', [104, 101, 108, 108, 111])
+    ];
+    var request = new http.Request.multipart(dummyUrl, files: files);
+
+    expect(request, bodyMatches('''
+        --{{boundary}}
+        content-type: application/octet-stream
+        content-disposition: form-data; name="file"
+
+        hello
+        --{{boundary}}--
+        '''));
+  });
+}
```

Comment:
Are these ported directly from the old API?

Reply:
Yep they're from the 0.11.x branch.